src/cratedb_fivetran_destination/engine.py

```diff
@@ -142,6 +142,50 @@ def column_definition(column):
         return f"{field} {type_}"
 
 
+@define
+class AlterTableRecreateStatements:
+    """
+    Manage and render a procedure of SQL statements for recreating a table with a new schema.
+
+    The procedure will create a new table, transfer data, and swap tables.
+    It is needed for propagating primary key column changes.
+    """
+
+    address_effective: TableAddress
+    address_temporary: TableAddress
+    columns_old: t.List[common_pb2.Column] = Factory(list)
+    columns_new: t.List[common_pb2.Column] = Factory(list)
+
+    def to_sql(self) -> SqlBag:
+        sqlbag = SqlBag()
+
+        table_real = self.address_effective.fullname
+        table_temp = self.address_temporary.fullname
+
+        # Put the source table into read-only mode at the beginning of the following operations.
+        sqlbag.add(f'ALTER TABLE {table_real} SET ("blocks.write"=true);')
+
+        # Define transform operation involving a temporary table copy and swap.
+        sqlbag.add(
+            f"""
+        INSERT INTO
+            {table_temp}
+            ({", ".join([f'"{FieldMap.to_cratedb(col.name)}"' for col in self.columns_new])})
+            (SELECT {", ".join([f'"{FieldMap.to_cratedb(col.name)}"' for col in self.columns_old])} FROM {table_real})
+        """  # noqa: S608, E501
+        )
+
+        # Put the source table into read+write mode again,
+        # and replace it with the new newly populated temporary table.
+        sqlbag.add(f'ALTER TABLE {table_real} SET ("blocks.write"=false);')
+        sqlbag.add(f"ALTER CLUSTER SWAP TABLE {table_temp} TO {table_real}")
+
+        # Drop the temporary table.
+        sqlbag.add(f"DROP TABLE {table_temp}")
```

Comment:
This wouldn't be needed when adding the `WITH (drop_source=true)` option to the previous `ALTER CLUSTER SWAP TABLE` statement. See https://cratedb.com/docs/crate/reference/en/latest/sql/statements/alter-cluster.html#options
```suggestion
```